tests/transforms/test_video_utils.py

```diff
@@ -55,27 +39,33 @@ def test_video_downsample(
     assert np.array_equal(downsampled_video, expected)
 
 
-def test_mp4_conversion(video_fixture, tmp_path):
+def test_mp4_conversion(tmp_path):
+    # make the test video a size of at least 16x16
+    # otherwise, need to mess with macro_block_size arg
+    nframes_written = 25
     output_path = tmp_path / 'video.mp4'
-
     fps = 30
+    rng = np.random.default_rng(0)
+    video = np.array(
+            [rng.integers(0, 256, size=(16, 16), dtype='uint8')
+             for i in range(nframes_written)])
 
-    transformations.transform_to_mp4(video=video_fixture,
+    transformations.transform_to_mp4(video=video,
                                      output_path=output_path.as_posix(),
                                      fps=fps),
 
-    frames = []
     reader = imageio.get_reader(output_path.as_posix(), mode='I', fps=fps,
-                                size=(2, 2), pixelformat="gray")
-    for frame in reader:
-        gray_frame = cv2.cvtColor(frame, cv2.COLOR_RGB2GRAY)
-        frames.append(gray_frame)
-    frames = np.stack(frames)
+                                pixelformat="gray")
+    meta = reader.get_meta_data()
+    nframes_read = int(np.round(meta['duration'] * meta['fps']))
+
+    assert nframes_read == nframes_written
 
-    # codec for mp4 encoding changes the video slightly between h5 and mp4
-    # and vice versa, so numbers are not exactly the same. Instead general
-    # structure of video array loaded. Cv color conversion does not rectify
-    # this error unfortunately, this has been attempted and small errors
-    # persist
+    read_video = np.zeros((nframes_read, *meta['size']), dtype='uint8')
+    for i in range(nframes_read):
+        read_video[i] = reader.get_data(i)[:, :, 0]
+    reader.close()
 
-    assert frames.shape == video_fixture.shape
+    assert read_video.shape == video.shape
+    # mp4 is not lossless compression, so can't test for exact match
```

Comment:
This is just a note

mp4 is merely the content container, the x264 codec that does the compression 'can' be lossless but you need to set an appropriate CRF:

https://slhck.info/video/2017/02/24/crf-guide.html

Reply:
changed comment to be more accurate